src/Shmuelie.WinRTServer/Internal/BaseActivationFactoryWrapper.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using Shmuelie.WinRTServer.Windows.Com;
+using Windows.Win32.Foundation;
+using static Windows.Win32.PInvoke;
+
+namespace Shmuelie.WinRTServer.Internal;
+
+[GeneratedComClass]
+internal sealed partial class BaseActivationFactoryWrapper(BaseActivationFactory factory, ComWrappers comWrappers) : IActivationFactory
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1031:Do not catch general exception types", Justification = "COM method, must not throw.")]
+    public unsafe HRESULT ActivateInstance(void** instance)
+    {
+        if (instance is null)
+        {
+            return HRESULT.E_INVALIDARG;
+        }
+
+        bool shouldReleaseUnknown = false;
+        nint unknown = 0;
+        try
+        {
+            object managedInstance = factory.ActivateInstance();
+            unknown = comWrappers.GetOrCreateComInterfaceForObject(managedInstance, CreateComInterfaceFlags.None);
+            var riid = global::Windows.Win32.System.WinRT.IActivationFactory.IID_Guid;
+            var hr = (HRESULT)Marshal.QueryInterface(unknown, ref riid, out nint ppv);

```

Comment:
You should be able to just pass `in IID_Guid` directly, no need for a temp local, nor passing via `ref`.